### PR TITLE
SLING-11745 - Sling Mocks: allow setting parts for a request

### PR DESCRIPTION
**Pull request by rombert**
Allow setting parts on a MockSlingHttpServletRequest and provide a simple ByteArrayPart implementation.

### CODE REVIEW

File: src/main/java/org/apache/sling/servlethelpers/ByteArrayPart.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+import javax.servlet.http.Part;
+
+/**
+ * Simple <tt>Part</tt> implementation backed by an in-memory byte array
+ *
+ */
+public class ByteArrayPart implements Part {
+	
+	/**
```
Comment (stefanseifert) [inline src/main/java/org/apache/sling/servlethelpers/ByteArrayPart.java:34]: this file is using tab indentation - please use 4 spaces

File: src/test/java/org/apache/sling/servlethelpers/ByteArrayPartTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+
+import org.junit.Test;
+
+public class ByteArrayPartTest {
+
+	private static final String TEST_CONTENT = "test input";
```
Comment (stefanseifert) [inline src/test/java/org/apache/sling/servlethelpers/ByteArrayPartTest.java:30]: this file is using tab indentation - please use 4 spaces

**Review by stefanseifert — APPROVED** [review]
lgtm, only two cosmetic findings

**Comment by rombert** [comment]
Converted tabs to spaces where I had accidentally introduced them, thanks for the review @stefanseifert

**Pull request merged**